src/write/elf/object.rs

```diff
@@ -327,6 +328,7 @@ impl<'a> Object<'a> {
         let e_machine = match self.architecture {
             Architecture::Aarch64 => elf::EM_AARCH64,
             Architecture::Aarch64_Ilp32 => elf::EM_AARCH64,
+            Architecture::Aarch64_Arm64e => elf::EM_AARCH64,
```

Comment:
According to https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#appendix-alternative-elf-marking-using-gnu-program-properties `GNU_PROPERTY_AARCH64_FEATURE_PAUTH` in `.note.gnu.property` indicates that pointer authentication is used.